### DISCUSSION TITLE
ERROR: Could not install packages due to an OSError: [Errno 28] No space left on device

### DISCUSSION THREAD

**Discussion by sebas1984x** [mcguirepr89/BirdNET-Pi#1270] [General]
i got this error when installing on a rpiw2, this is the solution.

The temporary directory (/tmp) i too small for the large package being installed.
To temporarily use another directory for TMPDIR, run:

export` TMPDIR=/home/pi/tmp
mkdir -p `/home/pi/tmp

Then re-run the installation.

**Comment by Award1000** [comment]
@sebas1984x - I did try the command `export TMPDIR=/home/pi/tmp mkdir -p /home/pi/tmp`? but I got the same error. I have tried reinstalling the OS Bookworm lite from scratch on the RPI02W, setting the swap file to a larger size and doing the above but I still get the same error "No space left on device". Any help appreciated!

The exact part of the script it falls over is:

Collecting requests-oauthlib
  Downloading https://www.piwheels.org/simple/requests-oauthlib/requests_oauthlib-2.0.0-py2.py3-none-any.whl (24 kB)
Collecting markdown
  Downloading https://www.piwheels.org/simple/markdown/Markdown-3.7-py3-none-any.whl (106 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 106.3/106.3 kB 1.6 MB/s eta 0:00:00
Collecting PyYAML
  Downloading PyYAML-6.0.2-cp311-cp311-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (736 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 736.8/736.8 kB 2.6 MB/s eta 0:00:00
Collecting attrs>=19.2.0
  Downloading https://www.piwheels.org/simple/attrs/attrs-24.3.0-py3-none-any.whl (63 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 63.4/63.4 kB 1.2 MB/s eta 0:00:00
Collecting iniconfig
  Downloading https://www.piwheels.org/simple/iniconfig/iniconfig-2.0.0-py3-none-any.whl (5.9 kB)
Collecting pluggy<2.0,>=0.12
  Downloading https://www.piwheels.org/simple/pluggy/pluggy-1.5.0-py3-none-any.whl (20 kB)
Collecting py>=1.8.2
  Downloading https://www.piwheels.org/simple/py/py-1.11.0-py2.py3-none-any.whl (98 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 98.7/98.7 kB 1.4 MB/s eta 0:00:00
Collecting tomli>=1.0.0
  Downloading tomli-2.2.1-cp311-cp311-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (226 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 226.1/226.1 kB 3.3 MB/s eta 0:00:00
Collecting audioread>=2.1.9
  Downloading https://www.piwheels.org/simple/audioread/audioread-3.0.1-py3-none-any.whl (23 kB)
Collecting scipy>=1.2.0
  Downloading scipy-1.15.1-cp311-cp311-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (38.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━      33.6/38.3 MB 3.9 MB/s eta 0:00:02
**ERROR: Could not install packages due to an OSError: [Errno 28] No space left on device**

**Comment by thomaszwagerman** [comment]
I also ran into this issue, but managed to fix it. As a note, I used this, better maintained version of BirdNET-Pi: https://github.com/Nachtzuster/BirdNET-Pi, but encountered the issue all the same. I have not fully diagnosed the underlying issue, but I have managed to get my installation to complete manually.

In my case, despite the installation "failing", birdnet will still run. Navigate to hostname.local (or your local ip address) and under View Log, it will tell you which packages are missing. I think perhaps the main culprit is attempts to install tensorflow (which is a large package), rather than the tflite_runtime-2.6.0 (the "lite" version of tensorflow). Maybe some or other package depends on tensorflow, and will attempt to import functions from it, rather than the lite version.

1. Open a terminal and ssh to your raspi (or navigate to Tools > Web Terminal).
2. In the BirdNET-pi folder, Activate the virtual env: `source birdnet/bin/activate`
3. Start with (my suspected culprit): `pip install tflite_runtime-2.6.0`. Any attempt to import tensorflow functions should now work.
4. Check back in the logs which other packages are missing and install these, one by one.
5. You may need to roll back your numpy version: `pip install numpy<2.0`

This should hopefully fix your packages issue. I had a subsequent issue of the labels not being detected. I assume this is normally handled for you in a subsequent step of the installation, but I had to manually unzip the label files.

Navigate to the model folder and unzip one of `labels_ln18.zip` or `labels_nm.zip` (I actually don't know what the difference is, if I'm honest?).

Change the name of your preferred language to `labels.txt`, in my case:
`mv labels_en.txt labels.txt`
